### PR TITLE
ChainSpace: Separate preparation of create chainspace extrinsic from chain dispatch Issue #197

### LINKED ISSUE #197: [C4GT] ChainSpace: Separate preparation of create chainspace extrinsic from chain dispatch
## Description
Separate chain-space creation preparation and dispatch of extrinsic to chain. This would help when bulk transaction takes place so we just prepare all the required extrinsics and then execute these transaction in batches by dumping them into a promise using 
a batch sign & submit utility ex, `authorizeBatch` which can take array of extrinsics, instead of just one.

## Goals

- [ ] Creation of new export function named `prepareCreateSpaceExtrinsic()`
- [ ] Integration of above method into existing `dispatchToChain()`

## Implementation Details
Developers/Contributors can refer to example PRs: #188, #182

---- 

## Product Name
CORD.JS

## Organization Name
Dhiway

## Domain
Blockchain, SDK Development, API

## Tech Skills Needed
Typescript 

## Mentor(s)
@amarts @NiranjanAP @vatsa287 @adi-a11y 

## Complexity
[Low]

## Category
[Performance Improvement], [Feature]

## Sub Category
 [API], [Refactoring], [Backend], [Beginner friendly]

### PR DESCRIPTION
**Pull request by hardiksharma11**
Fixes #197

### CODE REVIEW

**Comment by hardiksharma11** [comment]
@vatsa287 @amarts @NiranjanAP will you please review

File: packages/chain-space/src/ChainSpace.chain.ts [hunk 1]
```diff
@@ -263,6 +253,32 @@ export async function dispatchToChain(
...
+      tx,
+      signCallback,
+      authorAccount.address
+    )
+    return extrinsic;
+
+
+  } catch (error) {
+    throw new SDKErrors.CordDispatchError(
+      `Error preparing extrinsic for creation: "${error}".`
```
Comment (vatsa287) [inline packages/chain-space/src/ChainSpace.chain.ts]: nit: Error preparing extrinsic for creation of chainspace

**Comment by hardiksharma11** [comment]
@vatsa287  @amarts  @NiranjanAP  PTAL

**Review by vatsa287 — APPROVED** [review]
LGTM

File: packages/chain-space/src/ChainSpace.chain.ts (hunk 1 above)
Comment (hardiksharma11) [reply packages/chain-space/src/ChainSpace.chain.ts]: Done

**Review by vatsa287 — CHANGES_REQUESTED** [review]
Minor Comment:
Please move `prepare....` method above `dispatch...` method.

**Comment by hardiksharma11** [comment]
@vatsa287 done the required change

**Review by vatsa287 — APPROVED** [review]
LGTM

**Pull request merged**